mule-migration-tool-engine/src/main/java/com/mulesoft/tools/migration/report/html/HTMLReport.java

```diff
@@ -49,17 +47,18 @@
   private static final String MULESOFT_ICON = "icons/004_logo.svg";
   private static final String MULESOFT_ICON_TTF = "fonts/muleicons.ttf";
 
-  private final File reportDirectory;
   private ApplicationReport applicationReport;
   private ReportFileWriter reportFileWriter = new ReportFileWriter();
   private Configuration freemarkerConfig;
   private String runnerVersion;
 
-  public HTMLReport(List<ReportEntryModel> reportEntries, File reportDirectory, String runnerVersion) {
-    checkNotNull(reportEntries, "Report Entries cannot be null");
-    checkNotNull(reportDirectory, "Report directory cannot be null");
-    this.applicationReport = new ApplicationReportBuilder().withReportEntries(reportEntries).build();
-    this.reportDirectory = reportDirectory;
+  public HTMLReport(MigrationReport<ReportEntryModel> report, File reportDirectory, String runnerVersion) {
+    super(report, reportDirectory);
+
+    // sorting entries by key to allow better validation over the HTML
```

Comment:
If the previous order was given by the order the entries where reported it may make more sense to keep it.

Reply:
The order seems to change from run to run because I was running the tool and trying to use that output for the test, it was failing because of the order. Do you think is still better to maintain it?

Reply:
in that case let's keep your change

Reply:
I changed the way its done so instead of sorting just the entries for the HTML report they also have a deterministic order when they are generated (at the DefaultMigrationReport level). For some reason still the order that is generated by the runner is not the same as the one of the tests but I could fix that by changing the test expected file.